lib/parse.js

```diff
@@ -56,12 +61,151 @@ module.exports = {
+
+    // Checking for the all the required properties needed in a root file
+    if (!spec.hasOwnProperty('openapi')) {
+      return {
+        result: false,
+        reason: 'Specification must contain a semantic version number of the OAS specification'
+      };
+    }
+
+    if (!spec.hasOwnProperty('paths')) {
+      return {
+        result: false,
+        reason: 'Specification must contain Paths Object for the available operational paths'
+      };
+    }
+
+    if (!spec.hasOwnProperty('info')) {
+      return {
+        result: false,
+        reason: 'Specification must contain an Info Object for the meta-data of the API'
+      };
+    }
 
 
     // Valid specification
     return {
       result: true,
       openapi: spec
     };
+  },
+
+  getOasObject: function (file) {
+    let oasObj = file;
+
+    if (typeof oasObj === 'string') {
+      try {
+        oasObj = JSON.parse(oasObj);
+      }
+      catch (jsonEx) {
+        // Not direct JSON. Could be YAML
+        try {
+          oasObj = yaml.safeLoad(oasObj);
+        }
+        catch (yamlEx) {
+          // Not JSON or YAML
+          return {
+            result: false,
+            reason: 'The input must be valid JSON or YAML'
+          };
+        }
+        // valid YAML
+      }
+    }
+
+    return oasObj;
+  },
+
+  getRootFiles: function (filesPathArray) {
+    let rootFilesArray = [];
+
+    filesPathArray.forEach((filePath) => {
+      try {
+        let file = fs.readFileSync(filePath.fileName, 'utf8'),
+          oasObject = this.getOasObject(file);
+
+        if (this.validateRoot(oasObject).result) {
+          rootFilesArray.push(filePath.fileName);
+        }
+      }
+      catch (e) {
+        throw new Error(e);
+      }
+    });
+    return rootFilesArray;
+  },
+
+  resolveContent: function (openapi, options) {
+    return resolver.resolve(openapi, options.source, {
+      options: Object.assign({}, options),
+      resolve: true,
+      cache: [],
+      externals: [],
+      externalRefs: {},
+      rewriteRefs: true,
+      openapi: openapi
+    });
+  },
+
+  mergeFiles: function(source, options = {}) {
+    options.source = source;
+    options.origin = source;
+    return this.readSpecFile(source)
+      .then((content) => {
+        try {
+          return yamlParse.parse(content, { prettyErrors: true });
+        }
+        catch (err) {
+          throw new ReadError('\nLine: ' + err.linePos.start.line + ', col: ' +
+           err.linePos.start.col + ' ' + err.message);
+        }
+      }, (err) => {
+        throw new OpenError(err.message);
+      })
+      .then((unresolved) => {
+        if (options.resolve === true) {
+          return this.resolveContent(unresolved, options);
+        }
+      }, (err) => {
+        throw err;
+      })
+      .then((result) => {
+        return result.openapi;
+      }, (err) => {
+        throw err;
+      });
+  },
+
+  readSpecFile: function (file) {
+    if (file && file.startsWith('http')) {
```

Comment:
(starts with `http://` or `https://`) is a more complete check. This is prone to random cases where non-remote files have names starting with http - httpd.conf / http.json for example,